modules/developing/pages/astream-functions.adoc

```diff
@@ -229,229 +185,253 @@ Result::
+Optionally, you can declare a `logTopic` in the same way: `persistent://**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME**`.
+
+* `**DISPLAY_NAME**`: The function display name in {product}, such as `testpythonfunction`.
+
+. Use `pulsar-admin` to deploy the function to {product} or Pulsar using your configuration file:
 +
-[source,bash]
+[source,bash,subs="+quotes"]
 ----
-bin/pulsar-admin functions create --function-config-file  </absolute/path/to/your/javajar/func-create-config.yml>
+bin/pulsar-admin functions create --function-config-file **PATH_TO_FUNCTION_CONFIG_YAML**
 ----
++
+.Result
+[%collapsible]
+====
+A response of `Created Successfully!` indicates the function is deployed and ready to accept messages.
 
-. Check results: Go to the {astra_ui} to see your newly deployed function listed under the “Functions” tab for your Tenant. See <<controlling-your-function,Controlling your function>> for more information on testing and monitoring your function in {product_name}.
-
-.. You can also use the pulsar-admin command to list your functions:
+If the response is `402 Payment Required` with `Reason: only qualified organizations can create functions`, then you must upgrade to a xref:operations:astream-pricing.adoc[paid {product} plan].
+Organizations on the *Free* plan can use xref:streaming-learning:functions:index.adoc[transform functions] only.
+====
 +
-[source,bash]
+[TIP]
+====
+If your Python function contains only a single script and no dependencies, you can deploy the `.py` file directly, without packaging it into a `.zip` file or creating a configuration file:
+
+[source,bash,subs="+quotes"]
 ----
-bin/pulsar-admin functions list --tenant <mytenant> --namespace <my-namespace>
+$ ./pulsar-admin functions create \
+  --py **PATH_TO_PYTHON_FILE** \
+  --classname **FILE_NAME**.**CLASS_NAME** \
+  --tenant **TENANT_NAME** \
+  --namespace **NAMESPACE_NAME** \
+  --name **DISPLAY_NAME** \
+  --auto-ack true \
+  --inputs persistent://**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME** \
+  --output persistent://**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME** \
+  --log-topic persistent://**TENANT_NAME**/**NAMESPACE_NAME**/**TOPIC_NAME**
 ----
 
-== Add functions in {product_name} dashboard
-
-Add functions in the Functions tab of the Astra Streaming dashboard.
+If there is no class in the file, the `className` is only the filename without the extension.
+====
 
-. Select *Create Function* to get started.
-. Choose your function name and namespace.
+. Verify the deployment:
 +
-image::astream-name-function.png[Function and Namespace]
-
-. Select the file you want to pull the function from and which function you want to use within that file.
+[source,bash,subs="+quotes"]
+----
+bin/pulsar-admin functions list --tenant **TENANT_NAME** --namespace **NAMESPACE_NAME**
+----
++
```

Comment:
Define placeholders?

Reply:
They are defined in the prior steps, this just shows an alternative approach.